src/plugins/mf_priority.cpp

```diff
@@ -760,6 +948,8 @@ static void add_missing_bank_info (flux_plugin_t *p, flux_t *h, int userid)
     b->cur_active_jobs = 0;
     b->active = 1;
     b->held_jobs = std::vector<long int>();
+    b->projects = { "*" };
+    b->def_project = "*";
```

Comment:
When the flux-accounting info is missing, shouldn't the project be treated like the bank info, i.e. be somehow recorded as unknown? That is, if the bank info is missing, we don't know if the default project should be `"*"` since the real default should be set in the accounting database. In fact, should the default project be something encoded in the database and left our of the plugin itself (that way other sites could make their own decision what a default project should be, or if there should even be a default)?

Reply:
That's a great question. I can change this assignment to be set to indicate that the project for the user/bank's job is unknown until it gets its information - just re-ran the tests after changing these values to something like `"DNE"`, which is the same as the bank information, and I think everything still works as expected. 👍 

In the user table in the flux-accounting DB, the default project for any user/bank when being added is `*` (if one is not specified at user creation time; the default project can be changed later if desired):

```python
"""
projects         tinytext    DEFAULT '*'            NOT NULL    ON CONFLICT REPLACE DEFAULT '*',
default_project  tinytext    DEFAULT '*'            NOT NULL    ON CONFLICT REPLACE DEFAULT '*',
"""
```

and `*` is added as a valid project name to the DB when the DB is first created:

```python
    conn.execute(
        """
            CREATE TABLE IF NOT EXISTS project_table (
                project_id          integer    PRIMARY KEY AUTOINCREMENT,
                project             tinytext               NOT NULL,
                usage               real       DEFAULT 0.0 NOT NULL
            );"""
    )
    conn.execute("INSERT INTO project_table (project) VALUES ('*')")
```

Since this is the case, maybe we don't even need to assign values for `project` and `def_project` here? Removing these assignments should also be fine (at least, the the testsuite passes). What do you think? Sorry if I am misunderstanding you.

Reply:
Yes, that would look more correct to me. If the default is already set in the database, then the code should use that default and not a copy, since that might be redundant and also could cause the values to diverge if someone updates the database and forget to do the same in the plugin. This is why it is usually bad practice to copy a default value into more than one place. 

Of course, I'm not all that familiar with the code here, so maybe it made more sense in this situation, so I was just posing the question.

Reply:
just pushed up a change (really just dropped this commit) that added this assignment of a project/default project in the case where a user/bank submits a job before any information is loaded to the plugin.